R/ark.R

```diff
@@ -99,32 +154,37 @@ ark_chunk <- function(db_con, tablename, start = 1,
     query <- paste("SELECT * FROM", 
                    tablename, 
                    "WHERE rownum BETWEEN",
-                   (start - 1) * lines, 
+                   sql_integer((start - 1) * lines), 
                    "AND", 
-                   start * lines)    
+                   sql_integer(start * lines))    
   } else { 
     ## Any SQL DB can do offset
     query <- paste("SELECT * FROM", tablename, "LIMIT", 
-                   lines, "OFFSET", (start-1)*lines)
+                   sql_integer(lines), 
+                   "OFFSET", 
+                   sql_integer((start-1)*lines))
 
   }
   chunk <- DBI::dbGetQuery(db_con, query)
   
   append <- start != 1
-  
-  ext <- switch(compress,
-                "bzip2" = ".bz2",
-                "gzip" = ".gz",
-                "xz" = ".xz",
-                "none" = "",
-                ".bz2")
-  
   readr::write_tsv(chunk, 
-                   file.path(dir, paste0(tablename, ".tsv", ext)), 
+                   con, 
                    append = append)
 
 }
 
+## need to convert large integers to characters
+sql_integer <- function(x){
```

Comment:
Would that be an option?

``` r
sprintf("%.0f", 1e13)
#> [1] "10000000000000"
```

> Created on 2018-07-25 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

Reply:
yes! that's much nicer than messing with scipen option...